packages/realm/src/errors.ts

```diff
@@ -138,3 +140,45 @@ export class ClientResetError extends SyncError {
     };
   }
 }
+
+/**
+ * An error class that indicates that one or more object changes have been reverted by the server.
+ * This can happen when the client creates/updates objects that do not match any subscription, or performs writes on
+ * an object it didn't have permission to access.
+ */
+export class CompensatingWriteError extends SyncError {
+  /**
+   * The array of compensating writes performed by the server.
+   */
+  public infos: CompensatingWriteInfo[] = [];
```

Comment:
`info` is plural.  I do, however, think this name is a bit amiguous.  Perhaps, `writeErrors` would suffice.

Reply:
Ok, so as we talked about above, these aren't failed writes, they are actually writes performed after the server has determined they are not allowed.  In that case, probably just renaming the property to `writes` would be clear.

Reply:
Looks like @elle-j and I had the same idea

Reply:
I think a more specific description of what this array is is: `The array of information about each object that caused the compensating write.`. So I don't believe it's actually an array of writes, just the info about what caused the compensating write. Core's more verbose naming `CompensatingWriteErrorInfo` disambiguates this, but for our shorter field name, I think all of these work: `errorInfo`, `info` (since it's within the context of a `CompensatingWriteError`), `writeErrors`.

Reply:
Could also go with `revertedObjects` or something similar.

Reply:
https://github.com/realm/realm-js/issues/5609 created an issue for this, there's already too many comments on this PR 😁